src/americana.js

```diff
@@ -36,210 +21,7 @@ import SampleControl from "openmapsamples-maplibre/OpenMapSamplesControl.js";
-
-    lyrRoad.road.surface(),
-
-    lyrRail.rail.dashes(),
-    lyrRail.railService.dashes(),
-
-    lyrRail.narrowGauge.dashes(),
-    lyrRail.narrowGaugeService.dashes(),
-
-    lyrRail.lightRailTram.dashes(),
-    lyrRail.lightRailTramService.dashes(),
-
-    lyrRail.funicular.dashes(),
-
-    lyrRail.railway.fill(),
-
-    lyrOneway.road,
-    lyrOneway.link
-  );
-
-  layers.push(lyrBuilding.building);
-
-  var bridgeLayers = [
-    lyrRail.bridgeCasing,
-
-    lyrRoad.trunkLinkBridge.casing(),
-    lyrRoad.motorwayLinkBridge.casing(),
-
-    lyrRoad.roadLinkSimpleCasingBridge.casing(),
-
-    lyrRoad.tertiaryExpresswayBridge.casing(),
-    lyrRoad.secondaryExpresswayBridge.casing(),
-    lyrRoad.primaryExpresswayBridge.casing(),
-    lyrRoad.trunkBridge.casing(),
-    lyrRoad.motorwayBridge.casing(),
-
-    lyrRoad.roadSimpleCasingBridge.casing(),
-
-    lyrRoad.tertiaryLinkBridge.fill(),
-    lyrRoad.tertiaryLinkTollBridge.fill(),
-    lyrRoad.secondaryLinkBridge.fill(),
-    lyrRoad.secondaryLinkTollBridge.fill(),
-    lyrRoad.primaryLinkBridge.fill(),
-    lyrRoad.primaryLinkTollBridge.fill(),
-    lyrRoad.roadLinkSimpleFillBridge.fill(),
-    lyrRoad.motorwayLinkBridge.fill(),
-
-    lyrRoad.minorBridge.fill(),
-    lyrRoad.minorTollBridge.fill(),
-    lyrRoad.tertiaryBridge.fill(),
-    lyrRoad.tertiaryTollBridge.fill(),
-    lyrRoad.secondaryBridge.fill(),
-    lyrRoad.secondaryTollBridge.fill(),
-    lyrRoad.primaryBridge.fill(),
-    lyrRoad.primaryTollBridge.fill(),
-    lyrRoad.roadSimpleFillBridge.fill(),
-    lyrRoad.motorwayBridge.fill(),
-
-    lyrRoad.roadBridge.surface(),
-
-    lyrRail.railBridge.dashes(),
-    lyrRail.railServiceBridge.dashes(),
-
-    lyrRail.narrowGaugeBridge.dashes(),
-    lyrRail.narrowGaugeServiceBridge.dashes(),
-
-    lyrRail.lightRailTramBridge.dashes(),
-    lyrRail.lightRailTramServiceBridge.dashes(),
-
-    lyrRail.funicularBridge.dashes(),
-
-    lyrRail.railwayBridge.fill(),
-
-    lyrOneway.bridge,
-    lyrOneway.bridgeLink,
-  ];
-
-  layers.push(...lyrRail.getLayerSeparatedBridgeLayers(bridgeLayers));
-
-  layers.push(
-    //The labels at the end of the list draw on top of the layers at the beginning.
-    lyrWater.waterwayLabel,
-    lyrWater.waterLabel,
-    lyrWater.waterPointLabel,
-
-    lyrTransportationLabel.bridgeSpacer,
-    lyrTransportationLabel.label,
-
-    lyrPark.label,
-    lyrPark.parkLabel,
-    /* The ref label shows up at lower zoom levels and when the long name doesn't fit */
-    lyrAeroway.airportRefLabel,
-    lyrAeroway.minorAirportRefLabel,
-    lyrAeroway.airportLabel,
-    lyrAeroway.minorAirportLabel,
-    lyrAeroway.airportGate,
-
-    lyrHighwayShield.shield,
-
-    lyrHighwayExit.exits,
-
-    lyrPlace.state,
-    lyrPlace.village,
-    lyrPlace.town,
-    lyrPlace.city,
-    lyrPlace.countryOther,
-    lyrPlace.country3,
-    lyrPlace.country2,
-    lyrPlace.country1,
-    lyrPlace.continent
-  );
+  let layers = Layers.build();
```

Comment:
This is fine, I suppose, but if the goal is to generate a style JSON file headlessly, then we need to extract not only the surrounding `buildLayers()` function but also `buildStyle()` into a separate file.

Reply:
It was a bigger untangle to do that because of the code that accesses the URL bar and I've borked the refactor a couple times trying to do it.  Hence the staged approach so the diff doesn't get too unwieldy to review.

Reply:
Similar to the suggestion below about passing in a locale, `buildStyle()` can accept a base URL.